Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino

```diff
@@ -340,41 +327,25 @@ void doPublishOdom() {
   
   // Calculate the deltas since
   double dt = current_time.toSec() - last_odom_publish_time.toSec();
-  double delta_x = (vx * cos(th)) * dt;
-  double delta_y = (vx * sin(th)) * dt;
+  double delta_x = (vx * cos(pose.theta)) * dt;
+  double delta_y = (vx * sin(pose.theta)) * dt;
   double delta_th = vth * dt;
 
   // Update position
-  x += delta_x;
-  y += delta_y;
-  th += delta_th;
-    
-  // Broadcast odometry transform
-
-  // Since all odometry is 6DOF we'll need a quaternion created from yaw
-  geometry_msgs::Quaternion odom_quat = tf::createQuaternionFromYaw(th);
-  
-  // First, we'll publish the transform over tf
-  ros_odom_transform.header.stamp = current_time;
-  ros_odom_transform.transform.translation.x = x;
-  ros_odom_transform.transform.translation.y = y;
-  ros_odom_transform.transform.translation.z = 0.0;
-  ros_odom_transform.transform.rotation = odom_quat;
-  ros_odom_broadcaster.sendTransform(ros_odom_transform);
+  pose.x += delta_x;
+  pose.y += delta_y;
+  pose.theta += delta_th;
```

Comment:
Normalize the angle
http://docs.ros.org/api/angles/html/namespaceangles.html#a90ae8baf394c3ce775a3a14e48c7e7be
The arguments being convention to see angles +- 180 degrees and the possibility of overflowing?